collect_app/src/main/java/org/odk/collect/android/views/ODKView.java

```diff
@@ -520,4 +613,32 @@ private QuestionWidget getQuestionWidget(FormIndex formIndex) {
         }
         return null;
     }
+
+    /**
+     * Removes the widget and corresponding divider at a particular index.
+     */
+    public void removeWidgetAt(int index) {
+        int indexAccountingForDividers = index * 2;
+
+        // There may be a first TextView to display the group path. See addGroupText(FormEntryCaption[])
+        if (view.getChildCount() > 0 && view.getChildAt(0) instanceof TextView) {
+            indexAccountingForDividers += 1;
+        }
+        view.removeViewAt(indexAccountingForDividers);
+
+        if (index > 0) {
+            view.removeViewAt(indexAccountingForDividers - 1);
+        }
+
+        widgets.get(index).release();
+        widgets.remove(index);
+    }
+
+    public void setWidgetValueChangedListener(WidgetValueChangedListener listener) {
+        widgetValueChangedListener = listener;
+    }
+
+    public void widgetValueChanged(QuestionWidget changedWidget) {
+        widgetValueChangedListener.widgetValueChanged(changedWidget);
```

Comment:
I think you can add a null check as you did here https://github.com/opendatakit/collect/pull/3023/files#diff-e8349c3ebea4bd514a58e7ecf890cb19R704 just in case.